staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -131,6 +131,12 @@ type ClientConfigLoadingRules struct {
 	// WarnIfAllMissing indicates whether the configuration files pointed by KUBECONFIG environment variable are present or not.
 	// In case of missing files, it warns the user about the missing files.
 	WarnIfAllMissing bool
+
+	// WarnIfAllMissingCallback is the warning log callback to use in case of missing files.
+	warnIfAllMissingCallback func(string, ...interface{})
+
+	// warningMuted mutes the warning callback triggered in case of missing files
+	warningMuted bool
```

Comment:
is it necessary? in the case someone does not want to log anything, he can provide a WarnIfAllMissingCallback that does nothing.

Reply:
Since the behavior that we want to implement is: if we allocate a new `ClientConfigLoadingRules`, no matter how (if through ClientConfigLoadingRules or by allocating it explicitly), it must be defaulted to `klog.Warningf`, then how can we distinguish the following cases between each oher?
1. the `ClientConfigLoadingRules` has been created by `NewDefaultClientConfigLoadingRules`, then we muted the log by assigning nil to `WarnIfAllMissingCallback` (we do not want lo log anthing).
2. The `ClientConfigLoadingRules` has been allocated explicitly without setting anything for `WarnIfAllMissingCallback`
 (we want to log with the default function).
In the above cases `WarnIfAllMissingCallback==nil` has different meaning, then I needed a parameter to distinguish between each other.

Reply:
Ok sorry, I did not read carefully. If you think that it would be better to use that approach, I can implement this simpler mechanism:
* If `WarnIfAllMissingCallback` is nil, then  log to `klog.Warningf`
* else log to `WarnIfAllMissingCallback`.

And it is up to who uses `ClientConfigLoadingRules` to provide a method that does nothing. 
I think that the approach I proposed is clearer, but I understand that it can be too much code for a very limited functional change.